### PR TITLE
refactor(core): make zone error messages tree shakable

### PR DESCRIPTION
**Pull request by ramthir**
Make the error messages tree shakable from the production build to reduce the bundle size.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Error messages under the zone are not tree-shakable from the production build.

Issue Number: N/A

## What is the new behavior?
Error messages under the zone are made tree shakable from the production build.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/zone/ng_zone.ts [hunk 1]
```diff
@@ -159,13 +162,17 @@ export class NgZone {
 
   static assertInAngularZone(): void {
     if (!NgZone.isInAngularZone()) {
-      throw new Error('Expected to be in Angular Zone, but it is not!');
+      throw new RuntimeError(
+          RuntimeErrorCode.INVALID_ZONE,
```
Comment (AndrewKushnir) [inline packages/core/src/zone/ng_zone.ts:166]: I'd propose updating the error code a bit:
```suggestion
          RuntimeErrorCode.UNEXPECTED_ZONE_STATE,
```

**Review by AndrewKushnir — COMMENTED** [review]
@ramthir thanks for creating this PR 👍 I've left a small proposal for one of the error codes.

It looks like there is a [test](https://app.circleci.com/pipelines/github/angular/angular/47664/workflows/ae608165-f5b1-49e5-a88e-b10759141702/jobs/1184161) CI job failure related to the decreased bundle size for some test apps. Could you please update the payload sizes?

Thank you.

**Review by AndrewKushnir — APPROVED** [review]
@ramthir looks great, thanks for the help 👍

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by pkozlowski-opensource — APPROVED** [review]
LGTM

Reviewed-for: public-api 
Reviewed-for: size-tracking
Reviewed-for: fw-core

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 7da389a461f41632339c1212e7f2ee999638c0ce.

**Pull request closed without merging**